src/server/main.go

```diff
@@ -32,48 +33,59 @@ func getPort() string {
 	return "8080"
 }
 
+/*
+ For flag type interface:
+	- get the env variable using os.Getenv(key), will return a string
+	- identify what primitive type to convert the string to
+	- convert the string to that type
+	- set a flag variable with that type
+	- if string keep as string, no need to go through process
+*/
+
+func toBool(s string) bool {
+	boolVal, err := strconv.ParseBool(s)
+	if err != nil {
+		core.CheckErrorMessage(err, "Failed to convert string to bool:")
+	}
+	return boolVal
+}
+
 func main() {
 	var err error
 
-	// Parse flags
-	googleAuthConfig := flag.String("google_auth", "", "<path-to-config>; Enables google sign-in API login")
-	noAuthConfig := flag.Bool("no_auth", false, "Enables username-only authentication for testing and development")
-	userCsifConfig := flag.String("interface", "sqlite", "The storage interface")
-	sqlitePathConfig := flag.String("sqlitePath", "sql/sqlite", "The path to the sqlite working directory")
-	dsEmulatorHost := flag.String("ds_emu_host", "", "The emulator host address for cloud datastore")
-	dsProjectId := flag.String("ds_emu_project_id", "", "The gcp project id for the datastore emulator")
-	ipAddressConfig := flag.String("ip_address", "0.0.0.0", "IP address of server")
-	portConfig := flag.String("port", "8080", "Port to serve application, use \"auto\" to select the first available port starting at 8080")
+	// Flag for noAuthConfig to keep boolean type
+	noAuthConfig := flag.Bool("no_auth", toBool(os.Getenv("no_auth")), "Enables username-only authentication for testing and development")
```

Comment:
I like this strategy.  Checking `flag`s first and fallback on the environment variables.  This should work in prod just fine and be convenient for back-end dev too.